### PR TITLE
Implement missing router functions

### PR DESCRIPTION
**Pull request by mikemurray**
- Implement `Router.setQueryParam()`
- Fix `Router.getRouteName()`

### CODE REVIEW

**Comment by hrath2015** [comment]
Getting error when running tests... It is windows 10.
image

**Comment by hrath2015** [comment]
created PR #2396 to fix this.

File: imports/plugins/core/router/lib/router.js [hunk 1]
```diff
@@ -87,6 +88,32 @@ class Router {
     return current.query && current.query[name] || undefined;
   }
 
+  static setQueryParams(newParams) {
+    const current = Router.current();
+    const queryParams = Object.assign({}, current.query, newParams);
+
+    for (const key in queryParams) {
+      if (queryParams[key] === null || queryParams[key] === undefined) {
+        delete queryParams[k];
```
Comment (zenweasel) [inline imports/plugins/core/router/lib/router.js]: where is `k` coming from? Should that be `key`?

File: imports/plugins/core/router/lib/router.js [hunk 2]
```diff
@@ -87,6 +88,32 @@ class Router {
...
+      }
+    }
+
+    Router.go(current.route.name, current.params, queryParams);
+  }
+
+  static setQueryParams(newParams) {
```
Comment (zenweasel) [inline imports/plugins/core/router/lib/router.js]: Did you intend to have two implementations of `setQueryParams` here? Was maybe one supposed to be `setQueryParam`?

File: imports/plugins/core/router/lib/__tests__/router.spec.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  // TODO: Update test when server router is implemented to use Router.go()
+  Router.setCurrentRoute(Router.routes[0]);
+
+  const queryValue = Router.getQueryParam("filter");
+
+  expect(queryValue).toBe("red");
+});
+
+test("Route /test/:id param 'id' to equal '12345'", () => {
+  // TODO: Update test when server router is implemented to use Router.go()
```
Comment (mikemurray) [inline imports/plugins/core/router/lib/__tests__/router.spec.js]: Yes, it's relevant to every test. I can deduplicate if it's an issue.

**Pull request merged**